core/domain/suggestion_services.py

```diff
@@ -1826,3 +1828,929 @@ def update_question_suggestion(
+            stat.submitted_questions_count)
+        stats_model.accepted_questions_count = (
+            stat.accepted_questions_count)
+        stats_model.accepted_questions_without_reviewer_edits_count = (
+            stat.accepted_questions_without_reviewer_edits_count)
+        stats_model.first_contribution_date = stat.first_contribution_date
+        stats_model.last_contribution_date = stat.last_contribution_date
+
+    suggestion_models.QuestionContributionStatsModel.update_timestamps_multi(
+        stats_models_to_update,
+        update_last_updated_time=True)
+    suggestion_models.QuestionContributionStatsModel.put_multi(
+        stats_models_to_update)
+
+
+def _update_question_review_stats_models(
+    question_review_stats: List[
+        suggestion_registry.QuestionReviewStats]
+) -> None:
+    """Updates QuestionReviewStatsModel models for given question
+    review stats.
+
+    Args:
+        question_review_stats: list(QuestionReviewStats). A list of
+            QuestionReviewStats domain objects.
+    """
+    stats_dict = {}
+
+    for stat in question_review_stats:
+        stat_id = suggestion_models.QuestionReviewStatsModel.construct_id(
+            stat.contributor_user_id, stat.topic_id)
+        stats_dict[stat_id] = stat
+
+    stats_ids = stats_dict.keys()
+
+    # stats_dict.keys() will return KeysView[str] typed data and get_multi()
+    # function requires Sequence[Optional[str]]. Hence we will have to ignore
+    # type check for the following statement because making stats_id  to match
+    # Sequence[Optional[str]] will not be efficient.
+    stats_models_to_update_with_none = (
+        suggestion_models.QuestionReviewStatsModel.get_multi(stats_ids))  # type: ignore[arg-type]
+
+    stats_models_to_update: List[
+        suggestion_models.QuestionReviewStatsModel] = []
+    for stats_model in stats_models_to_update_with_none:
+        # We can confirm that stats_model will not be None since the for loop
+        # already gives the not None stats_model object. Hence we can rule out
+        # the possibility of None for mypy type checking.
+        assert stats_model is not None
+        stat = stats_dict[stats_model.id]
+        stats_model.reviewed_questions_count = (
+            stat.reviewed_questions_count)
+        stats_model.accepted_questions_count = (
+            stat.accepted_questions_count)
+        stats_model.accepted_questions_with_reviewer_edits_count = (
+            stat.accepted_questions_with_reviewer_edits_count)
+        stats_model.first_contribution_date = stat.first_contribution_date
+        stats_model.last_contribution_date = stat.last_contribution_date
+
+    suggestion_models.QuestionReviewStatsModel.update_timestamps_multi(
+        stats_models_to_update,
+        update_last_updated_time=True)
+    suggestion_models.QuestionReviewStatsModel.put_multi(
+        stats_models_to_update)
+
+
+def get_contribution_date(date: datetime.datetime) -> datetime.date:
```

Comment:
Should this be prepended with an underscore?

Reply:
Done